actix-web-httpauth/src/extractors/api_key.rs

```diff
@@ -0,0 +1,113 @@
+//! Extractor for the "Basic" HTTP Authentication Scheme.
+
+use std::borrow::Cow;
+
+use actix_utils::future::{ready, Ready};
+use actix_web::{dev::Payload, http::header::Header, FromRequest, HttpRequest};
+
+use super::{config::AuthExtractorConfig, errors::AuthenticationError};
+use crate::headers::{
+    api_key::{APIKey, XAPIKey},
+    www_authenticate::basic::Basic as Challenge,
+};
+
+/// [`BasicAuth`] extractor configuration used for [`WWW-Authenticate`] header later.
+///
+/// [`WWW-Authenticate`]: crate::headers::www_authenticate::WwwAuthenticate
+#[derive(Debug, Clone, Default)]
+pub struct Config(Challenge);
+
+impl Config {
+    /// Set challenge `realm` attribute.
+    ///
+    /// The "realm" attribute indicates the scope of protection in the manner described in HTTP/1.1
+    /// [RFC 2617 §1.2](https://tools.ietf.org/html/rfc2617#section-1.2).
+    pub fn realm<T>(mut self, value: T) -> Config
```

Comment:
I suspect this is not valid for API-Key auth.  However I see the `Bearer` extractor has `realm`, so maybe I am wrong.

The config that would be useful is what is the name of the header containing the api-key.  i.e. it shouldnt be hard-coded to `x-api-key`.
If not configurable, the docs for the module should indicate the limitations compared with https://swagger.io/docs/specification/authentication/api-keys/

Reply:
Re Bearer realm - it is allowed in the spec https://datatracker.ietf.org/doc/html/rfc6750#section-3

wrt X-API-Key, we should see if there is anyone using realm with this header